nuxeo-runtime/nuxeo-runtime-stream/src/main/java/org/nuxeo/runtime/avro/AvroComponent.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     pierre
+ */
+package org.nuxeo.runtime.avro;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.avro.Schema;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.common.xmap.annotation.XObject;
+import org.nuxeo.runtime.model.ComponentInstance;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * Avro component.
+ * 
+ * @since 10.2
+ */
+public final class AvroComponent
+        extends DefaultComponent {
+
+    private static final Log log = LogFactory.getLog(AvroComponent.class);
+
+    protected final AvroSchemaStoreService schemaStoreService;
+
+    public AvroComponent() {
+        schemaStoreService = new AvroSchemaStoreServiceImpl();
```

Comment:
Could you do this in an `activate` method so someone could use a different implementation of `AvroSchemaStoreService` if they want?

Reply:
How would they give the implementation class ?
I would have done so if there was the extension designed to do it, since there's not, doing this here allows a final field I'd prefer with no other gain in the balance.

Reply:
The Nuxeo approach is to design for extension, ie. favour non-final and public methods.  Your current code is fine but it specifically disallows extension.  If someone wants their own implementation of `AvroSchemaStoreService` they can't.  If the `schemaStoreService` is initiallised in a method it makes it easy to extend.

Reply:
In fact such extension mechanism exist, but unknown. From my point of view, instantiate if directly on field line declaration.

Reply:
@covolution that's a first version, we will consider and maybe design a way to specify a custom AvroSchemaStoreService implementation through an extension point. Until that you would like to be able to subclass the AvroComponent to force another implementation ?
I don't have any experience with Nuxeo usage, customer stories, etc. but I believe design for extension is different than allowing every and any thing.

Reply:
I am ok to keep it like you have it.

Reply:
I am ok to keep it like you have it.